app/config.c

```diff
@@ -122,14 +125,36 @@ static void parse_serial_params(const char *params)
         return;
     }
 
+    // Check if console is MMIO and grab address and stride
+    uintptr_t mmio_adr = 0;
+    if (strncmp(params, "mmio,0x", 7) == 0) {
```

Comment:
For consistency with the two other choices, I'd suggest using "mmio8,0x" and therefore a byte offset of 8 inside params. Also, "if(" typo below.

Reply:
Your point makes sense, but using `mmio` instead of `mmio8` is deliberate to mimic the Linux kernel command line parameters:

https://www.kernel.org/doc/html/v5.4/admin-guide/kernel-parameters.html

```
              ttyS<n>[,options]
                ttyUSB0[,options]
                        Use the specified serial port.  The options are of
                        the form "bbbbpnf", where "bbbb" is the baud rate,
                        "p" is parity ("n", "o", or "e"), "n" is number of
                        bits, and "f" is flow control ("r" for RTS or
                        omit it).  Default is "9600n8".

                        See Documentation/admin-guide/serial-console.rst for more
                        information.  See
                        Documentation/networking/netconsole.txt for an
                        alternative.

                uart[8250],io,<addr>[,options]
                uart[8250],mmio,<addr>[,options]
                uart[8250],mmio16,<addr>[,options]
                uart[8250],mmio32,<addr>[,options]
                uart[8250],0x<addr>[,options]
                        Start an early, polled-mode console on the 8250/16550
                        UART at the specified I/O port or MMIO address,
                        switching to the matching ttyS device later.
                        MMIO inter-register address stride is either 8-bit
                        (mmio), 16-bit (mmio16), or 32-bit (mmio32).
                        If none of [io|mmio|mmio16|mmio32], <addr> is assumed
                        to be equivalent to 'mmio'. 'options' are specified in
                        the same format described for ttyS above; if unspecified,
                        the h/w is not re-initialized.
```

Reply:
Oh, I didn't know. So indeed, compatibility beats consistency :)